lib/index.js

```diff
@@ -170,16 +170,107 @@ class VideoAsset extends BaseAsset {
 
 exports.VideoAsset = VideoAsset;
 
+function _ensure_cheerio(value) {
+    if (typeof value === 'string')
+        return cheerio.load(value);
+    return value;
+}
+
 class NewsArticle extends BaseAsset {
     constructor() {
         super();
         this._object_type = "ArticleObject";
         this._content_type = "text/html";
     }
 
-    set_document(value) { this._document = cheerio.load(value); }
+    // Takes a Date or something that a Date can be constructed out of
+    set_date_published(value) {
+        if (value instanceof Date)
+            this._date_published = value;
+        else
+            this._date_published = new Date(value);
+    }
+
+    // Takes a string or array of strings
+    set_authors(value) {
+        if (value instanceof Array)
+            this._authors = value;
+        else
+            this._authors = [value];
+    }
+
+    set_main_image(asset, caption) {
+        this._main_image = asset;
+        this._main_caption = _ensure_cheerio(caption);
+    }
+
     set_section(value) { this._section = value; }
     set_as_static_page() { this._ekn_tags.push(EKN_STATIC_TAG); }
+    set_source_name(value) { this._source_name = value; }
```

Comment:
The existing metadata property is called `sourceName`, so I thought it should, but no big deal.